### PR TITLE
KAFKA-3361: Initial protocol documentation page and generation (0.9)

### PR DESCRIPTION
**Pull request by granthenke**
- Moves all generated docs under /docs/generated
- Generates docs for Protocol, Errors, and ApiKeys
- Adds new protocol.html page

### CODE REVIEW

File: docs/protocol.html [hunk 1]
```diff
@@ -0,0 +1,163 @@
+<h3><a id="protocol" href="#protocol">Kafka Wire Protocol</a></h3>
+
+<p>This document covers the protocol implemented in Kafka 0.8 and beyond. It is meant to give a readable guide to the protocol that covers the available requests, their binary format, and the proper way to make use of them to implement a client. This document assumes you understand the basic design and terminology described <a href="https://kafka.apache.org/documentation.html#design">here</a></p>
```
Comment (gwenshap) [inline docs/protocol.html]: Didn't we change that?

Comment (granthenke) [inline docs/protocol.html]: We did...not sure why the cherry pick missed it. Will fix.

Comment (granthenke) [inline docs/protocol.html]: Updated. Tried to cherry pick both commits at once, must have ignored this one.

**Comment by gwenshap** [comment]
LGTM. Site repo is next? 
I'm not planning on reviewing it tonight though.

**Comment by granthenke** [comment]
@gwenshap Yeah, I will work on that tomorrow.

**Pull request closed without merging**